gcc/rust/ast/rust-ast-dump.cc

```diff
@@ -728,19 +847,28 @@ Dump::visit (RawPointerType &type)
 
 void
 Dump::visit (ReferenceType &type)
-{}
+{
+  type.get_type_referenced ()->accept_vis (*this);
+}
 
 void
 Dump::visit (ArrayType &type)
-{}
+{
+  type.get_elem_type ()->accept_vis (*this);
+}
 
 void
 Dump::visit (SliceType &type)
-{}
+{
+  type.get_elem_type ()->accept_vis (*this);
+}
 
 void
 Dump::visit (InferredType &type)
-{}
+{
+  // FIXME: Is that always valid?
+  stream << "_";
```

Comment:
What do you mean? Its fairly common to see it being used in interesting ways in the libcore code

Reply:
Just wondering if there are any other ways to format inferred types, but no. I'll remove that comment :)